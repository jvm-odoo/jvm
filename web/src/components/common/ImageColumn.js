import styled from 'styled-components'

import { Column } from './'

export const ImageColumn = styled(Column)`
  
`
