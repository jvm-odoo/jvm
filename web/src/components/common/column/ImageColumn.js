import React from 'react'
import styled from 'styled-components'

import { Column } from './'

const StyledImageColumn = styled(Column)`
  display: none;
  box-sizing: border-box;
  padding: 48px 24px;
  overflow: hidden;
  border-radius: 5px;
  position: relative;
  min-height: 60vh;
  box-shadow: 0px 0px 10px 0px rgba(28, 28, 28, 0.25);

  @media only screen and (min-width: 768px) {
    display: flex;
  }
`
const Overlay = styled.div`
  height: 100%;
  width: 100%;
  position: absolute;
  top: 0;
  left: 0;
  z-index: 2;
  background: linear-gradient(
    135deg,
    rgba(90,81,250 ,0.5) 0%,
    rgba(179,204,210 ,0.5) 100%,
    rgba(0,212,255 ,0.5) 100%
  );
`
const Image = styled.img`
  flex: 1;
  border-radius: 5px;
  position: absolute;
  top: -10vh;
  left: 0;
  z-index: 1;
`

export const ImageColumn = ({ background }) => {
  return (
    <StyledImageColumn padding>
      <Overlay />
      <Image src={background} alt='Jason Van Malder - Computer' />
    </StyledImageColumn>
  )
}
