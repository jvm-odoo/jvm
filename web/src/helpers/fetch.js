export const fetchGet = async (url) => {
  const response = await fetch(url)
  return response.json()
}
