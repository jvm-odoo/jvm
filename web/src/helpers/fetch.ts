export const fetchGet = async (url: string) => {
    const response = await fetch(url)
    return response.json()
}

export const fetchPost = async (url: string, data: any, token?: string) => {
    const headers: Headers = new Headers({
        'Content-Type': 'application/json',
        'Access-Control-Allow-Origin': '*'
    })

    if (token) {
        headers.append('Authorization', `Token ${token}`)
    }

    const response = await fetch(url, {
        method: 'POST',
        headers,
        mode: 'cors',
        body: JSON.stringify(data)
    })

    return response.json()
}