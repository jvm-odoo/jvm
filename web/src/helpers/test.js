import React from 'react'
import toJson from 'enzyme-to-json'
import { act } from 'react-dom/test-utils'
import { mount, shallow } from 'enzyme'
import { MockedProvider } from '@apollo/react-testing'

const wait = (time = 0) => new Promise(res => setTimeout(res, time))

export const executeMockProviderTestCase = async (wrapperInstance) => (
  await act(async () => {
    await wait(100)
    wrapperInstance.update()
  })
)

export const shallowMatchSnapshot = (Component) => {
  const wrapper = shallow(Component)
  expect(toJson(wrapper)).toMatchSnapshot()
}

export const mountWrapperWithMockedData = async (Component, mocks) => {
  let wrapper

  act(() => {
    wrapper = mount(
      <MockedProvider mocks={mocks} addTypename={false}>
        {Component}
      </MockedProvider>
    )
  })

  await executeMockProviderTestCase(wrapper)
  return wrapper
}
