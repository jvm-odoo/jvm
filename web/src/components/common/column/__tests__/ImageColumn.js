import React from 'react'
import { render } from 'enzyme'

import { ImageColumn } from '../ImageColumn'
import { shallowMatchSnapshot } from '../../../../helpers/test'

it('renders correctly', () => shallowMatchSnapshot(<ImageColumn />))

it('props propagated correctly', () => {
  const background = 'prout'
  const wrapper = render(<ImageColumn background={background}/>)
  expect(wrapper.find('img').prop('src')).toBe(background)
})

