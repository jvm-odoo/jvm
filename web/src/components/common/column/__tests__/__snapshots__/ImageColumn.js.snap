// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Styled(styled.div)
  padding={true}
>
  <styled.div />
  <styled.img
    alt="Jason Van Malder - Computer"
  />
</Styled(styled.div)>
`;
